internal/provider/table_schema_resource.go

```diff
@@ -299,14 +306,31 @@ func (t *tableSchemaResource) Delete(ctx context.Context, req resource.DeleteReq
 	}
 }
 
-func toPinotModelFieldSpec(fieldSpecs []fieldSpec) []model.FieldSpec {
+func boolPtr(b bool) *bool {
+	return &b
+}
```

Comment:
We actually are provided with a Bool Pointer Value type with the basetypes module for terraform:

```go
func toDimensionFieldSpecs(fieldSpecs []dimensionFieldSpec) []model.FieldSpec {

	var pinotFieldSpecs []model.FieldSpec
	for _, fs := range fieldSpecs {
		pinotFieldSpecs = append(pinotFieldSpecs, model.FieldSpec{
			Name:     fs.Name,
			DataType: fs.DataType,
			NotNull:  fs.NotNull.ValueBoolPointer(),
		})
	}
	return pinotFieldSpecs
}
```